test/tests/functional/pbs_hook_endjob.py

```diff
@@ -0,0 +1,1258 @@
+            self.endjob_delete_running_job_moms_stopped,
+            delete_force=True,
+            delete_user=TEST_USER,
+            job_rerunnable=False)
+
+    def test_hook_endjob_delete_running_single_job_as_root_sm(self):
+        """
+        Run a single rerunable job, but delete as user before completion
+        after stopping the MoM. Verify that the end job hook is executed.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            job_rerunnable=True)
+
+    def test_hook_endjob_delete_running_single_job_as_user_sm(self):
+        """
+        Run a single rerunable job, but delete as user before completion
+        after stopping the MoM. Verify that the end job hook is executed.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            delete_user=TEST_USER,
+            job_rerunnable=True)
+
+    def test_hook_endjob_force_delete_running_single_job_as_root_sm(self):
+        """
+        Run a single rerunable job, but force delete as root before completion
+        after stopping the MoM. Verify that the end job hook is executed.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            delete_force=True,
+            job_rerunnable=True)
+
+    def test_hook_endjob_force_delete_running_single_job_as_user_sm(self):
+        """
+        Run a single rerunable job, but force delete as user before completion
+        after stopping the MoM. Verify that the end job hook is executed.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            delete_force=True,
+            delete_user=TEST_USER,
+            job_rerunnable=True)
+
+    # FIXME: Possible server bug?  When a rerunnable job is deleted while the
+    # mom is stopped and restarted, the job is killed and requeued as expected.
+    # The job is rerun after the MoM is restarted and runs to completion;
+    # however, the job's substate indicates that it was deleted.
+    def test_hook_endjob_delete_running_single_job_as_root_rm(self):
+        """
+        Run a single rerunable job, but delete as root before completion after
+        stopping the MoM. Verify that the end job hook is executed.  Then
+        restart the mom and verify that the job is restarted and the end
+        job hook is executed again.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            restart_moms=True,
+            job_rerunnable=True)
+
+    # FIXME: Possible server bug?  When a rerunnable job is deleted while the
+    # mom is stopped and restarted, the job is killed and requeued as expected.
+    # The job is rerun after the MoM is restarted and runs to completion;
+    # however, the job's substate indicates that it was deleted.
+    def test_hook_endjob_delete_running_single_job_as_user_rm(self):
```

Comment:
Could the failure be related to the changes made to req_delete.c? Does the problem happen without the change? If this still looks like a PBS code bug, please file an OSS ticket. And rather than have this test fail, put a @skip(<message_why_skipped>) tag. For example:

    @skip("issue 2440")
    def test_hook_endjob_delete_running_single_job_as_user_rm(self):

Reply:
The failure scenario is described in issue #2485.  An `@skip` was added to the test in 0eaabbb7425b1a159e3dba3d2bd58c5976577ffd.